File: Source/Rio.Models/DataTransferObjects/Offer/PostingWithTradesWithMostRecentOfferDto.cs
using System;
using System.Collections.Generic;
using Rio.Models.DataTransferObjects.Posting;
using Rio.Models.DataTransferObjects.User;

namespace Rio.Models.DataTransferObjects.Offer
{
    public class PostingWithTradesWithMostRecentOfferDto
    {
        public int PostingID { get; set; }
        public PostingTypeDto PostingType { get; set; }
        public DateTime PostingDate { get; set; }
        public int Quantity { get; set; }
        public int AvailableQuantity { get; set; }
        public decimal Price { get; set; }
        public string PostingDescription { get; set; }

        public UserSimpleDto CreateUser { get; set; }
        public PostingStatusDto PostingStatus { get; set; }

        public List<TradeWithMostRecentOfferDto> Trades { get; set; }
    }
}

File: Source/Rio.Web/src/app/shared/models/posting/posting-with-trades-with-most-recent-offer-dto.ts
import { UserSimpleDto } from '../user/user-simple-dto';
import { PostingTypeDto } from './posting-type-dto';
import { PostingStatusDto } from './posting-status-dto';
import { TradeWithMostRecentOfferDto } from '../offer/trade-with-most-recent-offer-dto';

export class PostingWithTradesWithMostRecentOfferDto {
    PostingID: number;
    PostingType: PostingTypeDto;
    PostingStatus: PostingStatusDto;

    PostingDescription: string;
    Quantity: number;
    AvailableQuantity: number;
    Price: number;

    PostingDate: Date;

    CreateUser: UserSimpleDto;

    Trades: Array<TradeWithMostRecentOfferDto>;
    
    constructor(obj?: any) {
        Object.assign(this, obj);
    }
}

